### DISCUSSION TITLE
umi4 配置项 terserOptions不存在

### DISCUSSION THREAD

**Discussion by lbyupup** [umijs/umi#9257] [Q&A]
umi3中可以在`terserOptions`配置项中移除生产环境的log，umi4该配置好像已经删除，有大佬知道这个配置更换成什么了吗

**Comment by fz6m** [comment]
umi4 默认使用的是 esbuild 压缩器，配置方式：

```ts
// .umirc.ts

export default {
  jsMinifierOptions: {
    drop: ['console']
  }
}
```

若使用 terser ：

```ts
// .umirc.ts

export default {
  jsMinifier: 'terser',
  jsMinifierOptions: {
    compress: {
      drop_console: true
    }
  }
}
```

FYI :

 - [jsMinifierOptions](https://umijs.org/docs/api/config#jsminifieroptions)

**Reply by lbyupup** [reply]
好的，谢谢！

**Reply by DAOLIDEWONIU** [reply]
请问一下 怎么更改js输出目录 ？

**Reply by fz6m** [reply]
- [outputPath](https://umijs.org/docs/api/config#outputpath)
